network/src/protocols/health_checker/mod.rs

```diff
@@ -160,18 +164,21 @@ where
     fn handle_ping_request(
         &mut self,
         peer_id: PeerId,
-        ping_msg: Ping,
+        ping: u32,
```

Comment:
this definitely looks odd. I'd prefer if we kept the embedding for sake of readability. Pings are infrequent and should have no effect on perf. Change otherwise looks great!

Reply:
Good point. I added that into the code. I also wrapped up discovery and slapped it into this PR, let me know if that's an issue, if so, I can separate this into two PRs. Thanks!